api/server/sdk/volume.go

```diff
@@ -42,6 +45,29 @@ func (s *VolumeServer) driver(ctx context.Context) volume.VolumeDriver {
 	return s.server.driver(ctx)
 }
 
+func (s *VolumeServer) auditLog(ctx context.Context, method, format string, a ...interface{}) {
+	if s.server.auditLogWriter() != nil {
+		userinfo, ok := auth.NewUserInfoFromContext(ctx)
+		if !ok {
+			return
+		}
+		claims := &userinfo.Claims
+
+		log := logrus.New()
+		log.Out = s.server.auditLogWriter()
+		logger := log.WithFields(logrus.Fields{
+			"username": userinfo.Username,
+			"subject":  claims.Subject,
+			"name":     claims.Name,
+			"email":    claims.Email,
+			"roles":    claims.Roles,
+			"groups":   claims.Groups,
+			"method":   method,
+		})
+		logger.Infof(format, a...)
```

Comment:
Does this need to redefined for every auditLog call? Can instead cache a `logrus.New()` in `VolumeServer`  along with the auditLogWriter io.Writer?

Method could be added in the logger.Infof call.

Reply:
True. I'll have to create a VolumeServer.New() or something like that to initialize the object.

Reply:
talked about it offline, and we agreed that it was not necessary